### PR TITLE
Add 'date' command to mbed shell example.

### LINKED ISSUE #26: Define and provide implementation for SystemTime_support
#### Problem
The stubs are not present in Mbed OS port. The system needs an implementation for it. 

 #### Proposed Solution
Provide an implementation, behavior can be deduced from existing ports: [example](https://github.com/ARMmbed/connectedhomeip/blob/demo/src/platform/Zephyr/SystemTimeSupport.cpp).

### PR DESCRIPTION
**Pull request by lukgni**
#### Problem
We need some way to verify SystemTimeSupport implementation for Mbed-OS.

 #### Summary of Changes
Adding |date| command to shell example which may be used to display and set system date and time.
Update CMakeList.txt.

Usage example:
```
date set YYYY-MM-DD HH:MM:SS
date
```

Fixes #26
Preceding PR: #45

### CODE REVIEW

File: examples/shell/mbed/ChipShellCollection.h [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+#pragma once
+
+extern "C" {
+// A list of shell commands provided by ChipShell
+void cmd_base64_init(void);
+void cmd_btp_init(void);
+void cmd_device_init(void);
+void cmd_misc_init(void);
+void cmd_otcli_init(void);
+void cmd_mbed_utils_init(void);
```
Comment (ATmobica) [inline examples/shell/mbed/ChipShellCollection.h:27]: I'm not sure if we have to create ChipShellCollection.h file one more time. Maybe just create mbed-specific commands file i.e ChipShellMbedCollection.h with cmd_mbed_utils_init() and in main.cpp include both.

File: examples/shell/mbed/ChipShellCollection.h [hunk 2]
```diff
@@ -0,0 +1,28 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
```
Comment (pan-) [inline examples/shell/mbed/ChipShellCollection.h:3]:
```suggestion
 *    Copyright (c) 2021 Project CHIP Authors
```

File: examples/shell/mbed/cmd_mbed_utils.cpp [hunk 3]
```diff
@@ -0,0 +1,124 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
```
Comment (pan-) [inline examples/shell/mbed/cmd_mbed_utils.cpp:3]:
```suggestion
 *    Copyright (c) 2021 Project CHIP Authors
```

File: examples/shell/mbed/cmd_mbed_utils.cpp [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+    if (0 == argc)
+    {
+        if (Layer::GetClock_RealTimeMS(currTimeMS) != CHIP_SYSTEM_NO_ERROR)
+        {
+            streamer_printf(sout, "ERROR: Date/Time was not set\r\n");
+            return CHIP_ERROR_ACCESS_DENIED;
+        }
+
+        SecondsSinceEpochToCalendarTime(currTimeMS / 1000, year, month, dayOfMonth, hour, minute, second);
+        streamer_printf(sout, "%04" PRIu16 "-%02" PRIu8 "-%02" PRIu8 " %02" PRIu8 ":%02" PRIu8 ":%02" PRIu16 "\n\r", year, month,
```
Comment (pan-) [inline examples/shell/mbed/cmd_mbed_utils.cpp:67]: Why use a `PRIu16` for seconds at the end ?

File: examples/shell/mbed/main.cpp [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+{
+    // Initialize the default streamer that was linked.
+    const int rc = streamer_init(streamer_get());
+
+    if (rc != 0)
+    {
+        ChipLogError(Shell, "Streamer initialization failed: %d", rc);
+        return rc;
+    }
+
```
Comment (pan-) [inline examples/shell/mbed/main.cpp:42]: Should we enable the RTC in this function with `set_time(0)` ?

**Review by pan- — COMMENTED** [review]
Thanks @lukgni . This is very good to have the commands available.

**Review by fkjagodzinski — APPROVED** [review]
:+1:

File: examples/shell/mbed/cmd_mbed_utils.cpp (hunk 4 above)
Comment (lukgni) [reply examples/shell/mbed/cmd_mbed_utils.cpp]: Oh, it looks like typo - we should use PRIu8 here.

File: examples/shell/mbed/cmd_mbed_utils.cpp [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+
+    uint32_t newCurrTime = 0;
+    streamer_t * sout    = streamer_get();
+
+    int ret          = 0;
+    CHIP_ERROR error = CHIP_NO_ERROR;
+
+    VerifyOrExit(argc == 2, error = CHIP_ERROR_INVALID_ARGUMENT);
+
+    ret = sscanf(argv[0], "%" SCNu16 "-%02" SCNu8 "-%02" SCNu8, &year, &month, &dayOfMonth);
```
[the remark this answers is not in the document]
Comment (lukgni) [reply examples/shell/mbed/cmd_mbed_utils.cpp]: Done

File: examples/shell/mbed/cmd_mbed_utils.cpp [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+    if (3 != ret)
+    {
+        streamer_printf(sout, "ERROR: Date is in wrong format!\r\n");
+        ExitNow(error = CHIP_ERROR_INVALID_ARGUMENT;);
+    }
+
+    ret = sscanf(argv[1], "%02" SCNu8 ":%02" SCNu8 ":%02" SCNu8, &hour, &minute, &seconds);
```
[the remark this answers is not in the document]
Comment (lukgni) [reply examples/shell/mbed/cmd_mbed_utils.cpp]: Done.

File: examples/shell/mbed/cmd_mbed_utils.cpp (hunk 3 above)
Comment (lukgni) [reply examples/shell/mbed/cmd_mbed_utils.cpp]: Done.

File: examples/shell/mbed/ChipShellCollection.h (hunk 2 above)
Comment (lukgni) [reply examples/shell/mbed/ChipShellCollection.h]: Done.

File: examples/shell/mbed/ChipShellCollection.h (hunk 1 above)
Comment (lukgni) [reply examples/shell/mbed/ChipShellCollection.h]: Good idea! I updated my PR with this suggestion.

**Review by ATmobica — APPROVED** [review]

File: examples/shell/mbed/main.cpp (hunk 5 above)
Comment (lukgni) [reply examples/shell/mbed/main.cpp:43]: It looks like Mbed-Os will automatically execute `set_time(0)`  if the RTC is uninitialized and we call `gettimeofday()`function. Here is [source](https://github.com/ARMmbed/mbed-os/blob/4c4295225ec17f9485c61c72588a02f0d35d6b31/platform/source/mbed_rtc_time.cpp#L100).

Comment (pan-) [reply examples/shell/mbed/main.cpp:43]: Good point.

**Review by pan- — APPROVED** [review]
LGTM

**Pull request merged**